huggingface-gpt2/README.md

```diff
@@ -0,0 +1,156 @@
+Refer to the readme at [transformers](https://github.com/huggingface/transformers/tree/master/examples/language-modeling#language-model-training) for additional details.
+
+Download the data and export path as $DATA_DIR: 
+```bash
+    export DATA_DIR=/path/to/downloaded/data/
+```
+
+* TRAIN_FILE: `$DATA_DIR/wiki.train.tokens`
+* TEST_FILE: `$DATA_DIR/wiki.test.tokens`
+
+Below instructions refer to these hdf5 data files as the data to make accessible to training process.
+
+## GPT2 Language Modeling fine-tuning with ONNX Runtime in Azure Machine Learning
+
+1. Setup environment
+
+    * Transfer training data to Azure blob storage
+
+    To transfer the data to an Azure blob storage using [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest), use command:
+    ```bash
+    az storage blob upload-batch --account-name <storage-name> -d <container-name> -s $DATA_DIR
+    ```
+
+    * Register the blob container as a data store
+    * Mount the data store in the compute targets used for training
+
+    Please refer to the [storage guidance](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-access-data#storage-guidance) for details on using Azure storage account for training in Azure Machine Learning. 
+
+2. Build the docker image
+
+    Install the dependencies of the transformer examples and modified transformers into the base ORT Docker image.
+    ```bash
+    cd transformers
+    docker build --network=host -f docker/transformers-ort-gpu/Dockerfile . --rm --pull -t onnxruntime-gpt
+    cd ../..
+    ```    
+    - Push the image to a container registry. You can find additional [details](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-get-started-docker-cli) about tagging the image and pushing to an [Azure Container Registry](https://docs.microsoft.com/en-us/azure/container-registry/).
+    
+3. Execute finetuning
+
+    The GPT2 finetuning job in Azure Machine Learning can be launched using either of these environments:
+
+    * Azure Machine Learning [Compute Instance](https://docs.microsoft.com/en-us/azure/machine-learning/concept-compute-instance) to run the Jupyter notebook.
+    * Azure Machine Learning [SDK](https://docs.microsoft.com/en-us/python/api/overview/azure/ml/?view=azure-ml-py)
+
+    You will need a [GPU optimized compute target](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-set-up-training-targets#amlcompute) - _either NCv3 or NDv2 series_, to execute this pre-training job.
+
+    Execute the steps in the Python notebook [azureml-notebooks/run-finetuning.ipynb](azureml-notebooks/run-finetuning.ipynb) within your environment. If you have a local setup to run an Azure ML notebook, you could run the steps in the notebook in that environment. Otherwise, a compute instance in Azure Machine Learning could be created and used to run the steps.
+
+## GPT2 Language Modeling fine-tuning with ONNX Runtime locally
+
```

Comment:
Is there a minimum HW spec required or recommended for the local setup?

Reply:
This would depend on the support of the docker image. At the minimum, we can recommend the use of a system with at least one GPU, that supports CUDA 10.1.
@skaarthik what do you think?

Reply:
Added a statement specifying that the docker image has been tested on DGX2 and AML.